TSPL.docc/LanguageGuide/Distributed.md

```diff
@@ -0,0 +1,671 @@
+way of re-thinking of a problem, and the following examples should help you shift your thinking if you
+find yourself unsure how to handle a situation while moving from local-only programming, to adopting distributed actors.
+
+### Distributed actors generic over Actor Systems
+
+It is possible to declare a distributed actor with a generic actor system, however when doing so, it must declare a
+concrete `SerializationRequirement` that the actor system must require in order for this distributed actor to be
+compatible with it.
+
+For example, if we were to implement a distributed actor representing some common distributed algorithm, or just some
+simple computation like a `Greeter`, which does not rely on an specific transport specifics, we may declare it as
+follows:
+
+```swift
+distributed actor Greeter<ActorSystem>
+  where ActorSystem: DistributedActorSystem<any Codable> {
+  distributed func greet(name: String) -> String {
+    "Hello \(name)!"
+  }
+}
+```
+
+While the `ActorSystem` remains generic, we did have to specify
+what [`SerializationRequirement`](https://developer.apple.com/documentation/distributed/distributedactorsystem/serializationrequirement)
+the system should be using.
+This is necessary for the type-checking of distributed functions to remain in-tact, and prevents any attempts of using
+distributed actors with actor systems which cannot serialize all of their method's required parameter and result types.
+
+### Emulating callbacks
+
+While in-process actors are able to send and work with closures, this is not possible with distributed actors, as a
+closure
+cannot be serialized over the network. This is one difference in programming for distribution as opposed to local-only
+concurrency with actors.
+
+Let us consider the following "call me later" example expressed in pure local actors:
+
+```swift
+actor Alice {
+  func call(later: @Sendable (String) -> ()) async -> String {
+    Task.detached {
+      // do some asynchronous processing AFTER returning from the 'call' method...
+      later("Here's the info you asked for!")
+    }
+
+    return "Thanks for your call!"
+  }
+}
+```
+
+```swift
+actor Bob {
+  func test(alice: Alice) async {
+    let immediateReply = await alice.call { laterReply in
+      print("later: \(laterReply)")
+    }
+
+    print("immediately: \(immediateReply)")
+  }
+}
+
+// OUTPUT: 
+// immediately: Thanks for your call!
+// later: Here's the info you asked for!
+```
+
+This example models a situation where we call the `Alice` actor, and receive a reply immediately, but at the same time
+we registered a closure to be called at some later point in time when some additional information was looked up by the
+actor. In local-only actors, a common way of modeling this is passing a closure to the actor.
+
+In distributed actors, the same can be achieved using distributed method calls rather than closures, like this:
```

Comment:
We should explain that Alice and Bob must share the same ActorSystem. Also there are two cases: is Bob type known on Alice side or not